mmv1/products/networkservices/Gateway.yaml

```diff
@@ -0,0 +1,126 @@
+self_link: 'projects/{{project}}/locations/global/gateways/{{name}}'
+min_version: beta
+update_verb: :PATCH
+update_mask: true
+description: |
+  Gateway represents the configuration for a proxy, typically a load balancer.
+  It captures the ip:port over which the services are exposed by the proxy,
+  along with any policy configurations. Routes have reference to to Gateways
+  to dictate how requests should be routed by this Gateway.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  api: 'https://cloud.google.com/traffic-director/docs/reference/network-services/rest/v1beta1/projects.locations.gateways'
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+    timeouts: !ruby/object:Api::Timeouts
+      insert_minutes: 30
+      update_minutes: 30
+      delete_minutes: 30
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+autogen_async: true
+import_format: ['projects/{{project}}/locations/global/gateways/{{name}}']
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    min_version: beta
+    name: 'network_services_gateway_basic'
+    primary_resource_id: 'default'
+    vars:
+      resource_name: 'my-gateway'
+  - !ruby/object:Provider::Terraform::Examples
+    min_version: beta
+    name: 'network_services_gateway_advanced'
+    primary_resource_id: 'default'
+    vars:
+      resource_name: 'my-gateway'
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    required: true
+    url_param_only: true
+    description: |
+      Short name of the Gateway resource to be created.
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'selfLink'
+    description: |-
+      Server-defined URL of this resource.
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: 'createTime'
+    description: |
+      Time the AccessPolicy was created in UTC.
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: 'updateTime'
+    description: |
+      Time the AccessPolicy was updated in UTC.
+    output: true
+  - !ruby/object:Api::Type::KeyValuePairs
+    name: 'labels'
+    required: false
+    description: Set of label tags associated with the Gateway resource.
+  - !ruby/object:Api::Type::String
+    name: 'description'
+    required: false
+    description: |
+      A free-text description of the resource. Max length 1024 characters.
+  - !ruby/object:Api::Type::Enum
+    name: 'type'
+    required: true
+    description: |
+      The type of the customer-managed gateway. Possible values are: * OPEN_MESH * SECURE_WEB_GATEWAY.
+    values:
+      - :TYPE_UNSPECIFIED
+      - :OPEN_MESH
+      - :SECURE_WEB_GATEWAY
+  - !ruby/object:Api::Type::Array
+    name: 'ports'
```

Comment:
I see in the description that this field is immutable. Could you add `immutable:true` to it?

Reply:
Done. I also reviewed other fields with the same issue.